internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -669,3 +669,25 @@ func (r *Reconciler) cleanupDeployment(ctx context.Context, oldMSs []*clusterv1.
 
 	return nil
 }
+
+func (r *Reconciler) patchOldMSDeletionTimeoutsFromMD(ctx context.Context, oldMSs []*clusterv1.MachineSet, deployment *clusterv1.MachineDeployment) error {
+	log := ctrl.LoggerFrom(ctx)
+	for _, oldMS := range oldMSs {
+		patchHelper, err := patch.NewHelper(oldMS, r.Client)
+		if err != nil {
+			return errors.Wrapf(err, "failed to generate patch for MachineSet %q", klog.KObj(oldMS))
+		}
+
+		// Set all other in-place mutable fields that impact the ability to tear down existing machines.
+		oldMS.Spec.Template.Spec.NodeDrainTimeout = deployment.Spec.Template.Spec.NodeDrainTimeout
+		oldMS.Spec.Template.Spec.NodeDeletionTimeout = deployment.Spec.Template.Spec.NodeDeletionTimeout
+		oldMS.Spec.Template.Spec.NodeVolumeDetachTimeout = deployment.Spec.Template.Spec.NodeVolumeDetachTimeout
+
+		err = patchHelper.Patch(ctx, oldMS)
+		if err != nil {
+			log.Error(err, "Failed to update MachineSet", "MachineSet", klog.KObj(oldMS))
+			return errors.Wrapf(err, "failed to update MachineSet %q", klog.KObj(oldMS))
+		}
+	}
```

Comment:
```suggestion
	for _, oldMS := range oldMSs {
		patchHelper, err := patch.NewHelper(oldMS, r.Client)
		if err != nil {
			return errors.Wrapf(err, "failed to generate patch for MachineSet %q", klog.KObj(oldMS))
		}

		// Set all other in-place mutable fields that impact the ability to tear down existing machines.
		oldMS.Spec.Template.Spec.NodeDrainTimeout = deployment.Spec.Template.Spec.NodeDrainTimeout
		oldMS.Spec.Template.Spec.NodeDeletionTimeout = deployment.Spec.Template.Spec.NodeDeletionTimeout
		oldMS.Spec.Template.Spec.NodeVolumeDetachTimeout = deployment.Spec.Template.Spec.NodeVolumeDetachTimeout

		err = patchHelper.Patch(ctx, oldMS)
		if err != nil {
			return errors.Wrapf(err, "failed to update MachineSet %q", klog.KObj(oldMS))
		}
	}
```

returned error will be reported as reconciliation error.